geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/io/ConcatenatedFileWriter.scala

```diff
@@ -38,17 +38,30 @@ object ConcatenatedFileWriter {
       ipcOpts: IpcOption,
       sort: Option[(String, Boolean)],
       files: CloseableIterator[Array[Byte]]): CloseableIterator[Array[Byte]] = {
-    // ensure we return something
-    if (files.hasNext) { files } else {
-      val dictionaries = dictionaryFields.mapWithIndex { case (name, i) =>
-        name -> ArrowDictionary.create(sft.getTypeName, i, Array.empty[AnyRef])
+    // NB: This is not a WithClose situation.
+    //  If there is an empty/non-exceptional iterator, we wish to use it.
+    //  If there are any issues, we wish to close the iterator to free memory.
+    try {
+      if (files.hasNext) {
+        files
+      } else {
+        generateEmptyResponse(sft, dictionaryFields, encoding, ipcOpts, sort)
```

Comment:
don't we still need to close `files` in this case?

Reply:
Good question.  Likely a good idea...  That said, the one calling site is...
```
    override def apply(features: CloseableIterator[SimpleFeature]): CloseableIterator[SimpleFeature] = {
      val bytes = features.map(_.getAttribute(0).asInstanceOf[Array[Byte]])
      val result = ConcatenatedFileWriter.reduce(sft, dictionaryFields, encoding, ipcOpts, sort, bytes)
```
Since we've mapped over a CloseableIterator, if it is empty, then it has closed ```features```, right?

Reply:
Oh wait, I think I'm wrong.  I don't see the code auto-closing...